users/app.py

```diff
@@ -22,31 +22,27 @@
 """
 
 import os
-
 from flask import Flask, jsonify, request
+import pymongo
 
 app = Flask(__name__)  # pylint: disable=invalid-name
 
+DB = None  # db initialized in main runtime
```

Comment:
Have you tested this with `gunicorn` or running in a container?

I think that `gunicorn` imports your module but it's no longer named `__main__`, so you might need to either use a function to initialize this globally, or add an `@app.before_first_request` method that will lazy-initialize on the first request.

(Or, I did more work than I needed last year when I was doing something like this with Cloud Functions).

Reply:
You're right, it doesn't initialize when deployed to Cloud Run. The reason I wasn't initializing this globally is because Travis-CI protects environmental secrets from pull requests from forks, so in order to get the tests the run I needed the app to not try to connect to the database.

I can think of 2 ways to solve this: @app.before_first_request as you suggested which probably works with few changes but looks a little fancy or refactoring the helper functions from app.py to some other imported module and only testing that module which maybe is better for organization but sounds like a pain and needless complication. Do you have a suggestion for which path to take?

Reply:
Fixed using @app.before_first_request that lazily inits a global DB object.